packages/cli/src/dbt/models.ts

```diff
@@ -98,36 +104,41 @@ const askOverwriteDescription = async (
     if (overwrite) return newDescription;
     return existingDescription;
 };
-export const updateModelYmlFile = async ({
+
+type FindAndUpdateModelYamlArgs = {
+    model: CompiledModel;
+    table: WarehouseTableSchema;
+    docs: Record<string, Doc>;
+    spinner: ora.Ora;
+};
+export const findAndUpdateModelYaml = async ({
     model,
     table,
     docs,
     spinner,
-}: UpdatedModelYmlFileArgs): Promise<{
+}: FindAndUpdateModelYamlArgs): Promise<{
     updatedYml: YamlSchema;
     outputFilePath: string;
 }> => {
-    const generatedModel = {
-        name: model.name,
-        columns: Object.entries(table).map(([columnName]) => ({
-            name: columnName,
-            description: '',
-        })),
-    };
-    if (model.patchPath) {
-        const { path: yamlSubpath } = patchPathParts(model.patchPath);
-        const outputFilePath = path.join(model.rootPath, yamlSubpath);
-        const existingYml = await loadYamlSchema(outputFilePath);
-        const models = existingYml.models || [];
-        const existingModelIndex = models.findIndex(
-            (m) => m.name === model.name,
-        );
-        if (existingModelIndex === -1) {
-            throw new ParseError(
-                `Expected to find model ${model.name} in ${outputFilePath} but couldn't find it`,
-            );
-        }
-        const existingModel = models[existingModelIndex];
+    const generatedModel = generateModelYml({ model, table });
+    const filenames = [];
+    const { patchPath } = model;
+    if (patchPath) {
+        const { path: expectedYamlSubPath } = patchPathParts(patchPath);
+        const expectedYamlPath = path.join(model.rootPath, expectedYamlSubPath);
+        filenames.push(expectedYamlPath);
+    }
+    const defaultYmlPath = path.join(
+        path.dirname(path.join(model.rootPath, model.originalFilePath)),
+        `${model.name}.yml`,
+    );
+    filenames.push(defaultYmlPath);
+    const match = await searchForModel({
+        modelName: model.name,
+        filenames,
+    });
+    if (match) {
+        const existingModel = match.doc.models[match.modelIndex];
```

Comment:
You could return the model instead of the index so you don't have to access the model again...

Reply:
I did originally! But later we re-insert the model at the original index in the yaml file, so it's useful to know where it lives!